test/t8_schemes/t8_gtest_face_descendant.cxx

```diff
@@ -0,0 +1,187 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2010 The University of Texas System
+  Written by Carsten Burstedde, Lucas C. Wilcox, and Tobin Isaac
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#include <gtest/gtest.h>
+#include <t8_eclass.h>
+#include <t8_schemes/t8_default/t8_default_cxx.hxx>
+
+/*TODO: Delete GTEST_SKIP in TEST_P t8_check_face_desc if t8_prism_children_at_face is fixed. */
+
+/* *INDENT-OFF* */
+class class_descendant:public testing::TestWithParam <t8_eclass_t > {
+protected:
+  void SetUp () override {
+    eclass = GetParam();
+
+    scheme = t8_scheme_new_default_cxx ();
+    /* Get scheme for eclass */
+    ts = scheme->eclass_schemes[eclass];
+  }
+  void TearDown () override {
+    /* Destroy scheme */
+    t8_scheme_cxx_unref (&scheme);
+  }
+  t8_eclass_t         eclass;
+  t8_scheme_cxx      *scheme;
+  t8_eclass_scheme_c *ts;
+
+};
+/* *INDENT-ON* */
+
+/* Computes manually the test_child_id-th child. */
+void
+t8_face_descendant_test_child (t8_element_t *tmp, t8_element_t *test,
+                               t8_element_t *elem, t8_eclass_scheme_c *ts,
+                               int face, int level, int ilevel,
+                               int num_children, int test_child_id)
+{
+
+  int                *child_indices = T8_ALLOC (int, num_children);
+
+  t8_element_t      **children = T8_ALLOC (t8_element_t *, num_children);
+  ts->t8_element_new (num_children, children);
+
+  ts->t8_element_copy (elem, tmp);
+  for (int klevel = level; klevel < ilevel; klevel++) {
+    /* Compute child_id of the test_child_id-th child. */
+    ts->t8_element_children_at_face (tmp, face, children, num_children,
+                                     child_indices);
+    int                 child_id = child_indices[test_child_id];
+
+    ts->t8_element_child (tmp, child_id, tmp);
+  }
+  ts->t8_element_destroy (num_children, children);
+  T8_FREE (children);
+  T8_FREE (child_indices);
+}
+
+/* Check the linear first and last descendants of an element along all faces. */
+void
+t8_linear_face_descendant (t8_element_t *elem, t8_element_t *tmp,
+                           t8_element_t *test, t8_eclass_scheme_c *ts,
```

Comment:
`elem` can be const.
`tmp` should be called something to indicate that it is the manual comparison computation, and `test` is the scheme calculation.